### ISSUE TITLE
MySQL 8.0 support

### ISSUE THREAD

**Issue by rur0** [AuthMe/AuthMeReloaded#1779]
### What behaviour is observed:
Server crashes without starting when using MYSQL storage with MYSQL 8 server

### What behaviour is expected:
Server to start and run authme smoothly

### Steps/models to reproduce:
enable and configure MYSQL with a MYSQL 8 server

### Plugin list:
ProtocolLib, SkinsRestorer, AuthMe

### Environment description
PaperMC standalone server with AuthMe MYSQL storage

### AuthMe build number:
AuthMe v5.5.1-SNAPSHOT-b2214

### Error Log:
https://pastebin.com/D4afV6fP

### Configuration:
https://pastebin.com/gM6SyHg6

edit: Also, when I am trying to compile AuthMe, I get this error from maven http://i.imgur.com/osz9Nik.png am I missing some repository ?

**Comment by sgdc3** [comment]
Spigot's built-in mysql connector doesn't support mysql 8 sadly,, we need to include our own connector library to fix this

**Comment by ruro1** [comment]
So in the meantime I can use any mysql version that is not 8. Or is there a recommended version ?

**Comment by ljacqu** [comment]
Yes

**Comment by games647** [comment]
You could also use MariaDB. It's a fork of MySQL and largely replaced the mysql package in many Linux distributions. However MariaDB 10+ is a major release and it could also contain breaking changes like in MySQL 8 from 5.7.

**Comment by games647** [comment]
BTW I noticed that the MySQL connector Spigot currently uses [should support MySQL 8](https://github.com/mysql/mysql-connector-j/commit/cfc78706659b0c0cd37a1ce18d7d637c09717e5b). The fix should be included in Connector 5.1.44. This is available in Spigot [1.12.2+](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/9a1f5ee80a8665b0c388758003e2bda1ee924084#pom.xml)

**Comment by tommasobenatti** [comment]
I'm having this problem too with the latest build downloaded from the CI, please fix

**Comment by sgdc3** [comment]
@tommasobenatti which MC version is your server at?

**Comment by tommasobenatti** [comment]
> @tommasobenatti which MC version is your server at?

Paper 1.8.8, 
with 1.12.2 there isn't this problem

**Comment by sgdc3** [comment]
@tommasobenatti please try using the "legacy" jar here: https://ci.codemc.io/job/AuthMe/job/AuthMeReloaded/

**Comment by sgdc3** [comment]
@tommasobenatti @rur0 Can somebody please test the legacy jar?

**Comment by tommasobenatti** [comment]
Sorry, at the moment I'm very busy and I can't test

**Comment by r3cord** [comment]
@sgdc3 
I've tested legacy and normal version on tuinity 1.16.5. It doesn't work with MariaDB 10
https://pastebin.com/Tf3igqeS

**Comment by tommasobenatti** [comment]
I think that you should test on a 1.8 server

**Comment by krusic22** [comment]
Their issue is related to #2303 and 1.8 is obsolete anyway.

**Comment by r3cord** [comment]
@tommasobenatti ohh so there is no fix for this problem in not legacy version of the plugin?

**Comment by tommasobenatti** [comment]
This issue was created because AuthMe wasn't working on 1.8 and MySQL 8 afaik

**Comment by tommasobenatti** [comment]
Since 1.8 is older than MySQL 8 and it dosn't include the library to connect to the database.
In fact the plugin works fine with 1.12 and MySQL 8

**Comment by r3cord** [comment]
@krusic22 Yeah you're right. Thank you very much! I've had to change com.mysql.jc.jdbc.Driver to com.mysql.jdbc.Driver and now it works.
@tommasobenatti yeah my fault, this thread wasn't correlating to my problem. Sorry!

**Comment by NanoNM** [comment]
Modify POM 

<dependency>
    <groupId>mysql</groupId>
    <artifactId>mysql-connector-java</artifactId>
    <version>8.0.xx</version>
    <optional>true</optional>
</dependency>

package jar 

Modify config.yml
mySQLDriverClassName: com.mysql.cj.jdbc.Driver
mySQLDatabase: xxx?serverTimezone=xxx

I use this method

**Comment by RuofengX** [comment]
## Version
Paper 1.16.5 build777
AuthMeReloaded v5.6.0-beta2 (build: 2453)
MYSQL 8
Ubuntu 2004
## Error
Sometimes startup with ERROR, especially after starting up the server or restarting mysql:


## Solution
in MYSQL:
`ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'Your New Password';`
guided by this post: [https://qiita.com/shaching/items/5fe3d5df691b4ec53084](https://qiita.com/shaching/items/5fe3d5df691b4ec53084)
